contracts/Database.sol

```diff
@@ -43,9 +51,41 @@ contract Database {
         return _id;
     }
 
+    function _checkGateName(string memory _name) private view {
+        bytes32 _nameId = keccak256(bytes(_name));
+        uint256[] memory gateIds = addressGates[msg.sender];
+        for (uint256 index = 0; index < gateIds.length; index++) {
+            Gate memory _gate = gates[gateIds[index]];
+            require(keccak256(bytes(_gate.name)) != _nameId, string(abi.encodePacked(_name, "is already taken by another gate")));
+        }
+    }
+
+    function deleteGate(uint256 _gateId) public virtual {
```

Comment:
Make sure only the owner of the gate can delete the gate

Reply:
this is already the case but I have added a test to ensure that it works